STYLE_GUIDE.md

```diff
@@ -0,0 +1,152 @@
+# KodaDot Style Guide v1.0
+
+As codebases grow in size and complexity, it is necessary to establish and maintain some kind of **style guide which each and every contributor should conform to**. While tools like ESLint and Prettier are able to support us in this regard, additional rules are needed to allow us to better communicate with each other. Since our community is coming from all around the world, working together in an asynchronous manner, **clear communication** is  of utmost importance.
+
+The following set of conventions should make it easier for you to understand our code and aid you in making meaningful contributions to the project:
+
+## Naming Conventions
+Give your functions, components and files **speaking/self-explanatory names** (e.g. `getCollectionById()`, `KeyboardShortcuts.vue`, `setIdentity.ts`).
+With a few exceptions, code and comments should be written in **english** only.
+
+### Nuxt/Vue-specific Files
+- **Pages** and **layouts** use **kebab-case** (`series-insight.vue`)
+- **All other components** (in folder `/components`) use **PascalCase** (`CookieBanner.vue`)
+
+### Others
+- **Folders** are written in **lowercase** (`components/spotlight`)
+- **Typescript**, **Javascript** and **GraphQL** files use **camelCase** (`globalVariables.ts`, `getKey.js`, `collectionById.graphql`)
+- **SCSS** files use **kebab-case** (`initial-variables.scss`)
+- **JSON** files use **snake_case** (`all_lang.json`) while **Markdown** files use **SCREAMING_SNAKE_CASE** (`CONTRIBUTING.md`)
+
+## SFC Conventions
+### Skeleton
+99% of the time your SCSS should be **scoped**, so it won't bleed outside of your component and pollute the global namespace!
+```vue
+<template>
+    <div>
+        ...
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'nuxt-property-decorator'
+import { ComponentOne } from '@/components/ComponentOne.vue'
+
+interface InterfaceName {
+  property: type
+}
+
+@Component({
+  components: {
+    ComponentOne
+  },
+})
+export default class ComponentName extends Vue {}
+</script>
+
+<style scoped lang="scss"></style>
+```
+
+### Property Decorators
+We rely on the package 'nuxt-property-decorator', hence, we urge you to comply with the [Nuxt Class Component Syntax](https://github.com/nuxt-community/nuxt-property-decorator/)
+```typescript
+import {
+  Component,
+  Inject,
+  Model,
+  Prop,
+  Provide,
+  Vue,
+  Watch,
+} from "nuxt-property-decorator"
+
+@Component({})
+export class MyComponent extends Vue {
+  @Inject() foo!: string
+
+  @Model("change") checked!: boolean
+
+  @Prop() propA!: number
+
+  @Provide() foo = "foo"
+
+  @Watch("child")
+  onChildChanged(val: string, oldVal: string) {}
+}
+```
+
+
+### Using Components In Templates
+Custom components and prop bindings should be used like this
+```vue
+<MyCustomComponent :property-name="doSomething">
```

Comment:
```suggestion
<MyCustomComponent :propertyName="doSomething">
```

I prefer camel case since I were coding React inb4 and also it's easier to distinct between native html5 attributes and component ones